docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Add-Providers-Support
+authors:
+  - "@Gal-Zaidman"
+creation-date: 2021-07-25
+last-updated: 2021-07-25
+---
+
+# Add Providers Support
+
+## Summary
+
+Currently, the assisted installer supports Bare-Metal installations only.
+Openshift can be installed on various providers such as:
+- Red Hat Virtualization.
+- OpenStack.
+- vSphere.
+- AWS.
+- GCP.
+- Azure.
+
+This enhancement will define a clear interface to extend the assisted installer and allow
```

Comment:
Somewhere in this proposal, it would be valuable to elaborate on what kinds of provider-specific behaviors will be addressed. Including a list of all such behaviors we're aware of so far would be useful. That'll give people a more tangible sense of what to expect.

Reply:
I aimed for the [provider interface](https://github.com/openshift/assisted-service/pull/2272/files#diff-988c4e5d655472b2174837171f5d89978ca0f86ec8554329b386dccb9990ea83R84-R90) to do that.
I would expand it and update the enhancement as we will find more provider-specific parts, in the meantime, I relaied on the adding vSphere PR that was merged by the team